hathor/verification/nano_header_verifier.py

```diff
@@ -115,3 +124,41 @@ def verify_action_list(actions: Sequence[NCAction]) -> None:
             action_types = {action.type for action in actions_per_token}
             if action_types not in ALLOWED_ACTION_SETS:
                 raise NCInvalidAction(f'conflicting actions for token {token_uid.hex()}')
+
+    def verify_method_call(self, tx: BaseTransaction) -> None:
+        assert tx.is_nano_contract()
+        assert isinstance(tx, Transaction)
+
+        nano_header = tx.get_nano_header()
+        if nano_header.is_creating_contract():
+            # TODO Confirm that the blueprint exists.
+            return
+
+        best_block = self._tx_storage.get_best_block()
+        block_storage = self._tx_storage.get_nc_block_storage(best_block)
+        contract_storage = block_storage.get_contract_storage(nano_header.nc_id)
+        blueprint_id = contract_storage.get_blueprint_id()
+        blueprint_class = self._tx_storage.get_blueprint_class(blueprint_id)
+        method_name = nano_header.nc_method
+        method = getattr(blueprint_class, method_name)
+        if method is None:
+            fallback_method = getattr(blueprint_class, NC_FALLBACK_METHOD, None)
+            if fallback_method is None:
+                raise NCMethodNotFound(f'method `{method_name}` not found and no fallback is provided')
+        else:
+            if not is_nc_public_method(method):
+                raise NCInvalidMethodCall(f'method `{method_name}` is not a public method')
+            parser = Method.from_callable(method)
+            parser.deserialize_args_bytes(nano_header.nc_args_bytes)
+
+    def verify_seqnum(self, tx: BaseTransaction) -> None:
+        assert tx.is_nano_contract()
+        assert isinstance(tx, Transaction)
+
+        nano_header = tx.get_nano_header()
+        best_block = self._tx_storage.get_best_block()
+        block_storage = self._tx_storage.get_nc_block_storage(best_block)
+        seqnum = block_storage.get_address_seqnum(nano_header.nc_address)
+        diff = nano_header.nc_seqnum - seqnum
+        if diff < 0 or diff > MAX_SEQNUM_JUMP_SIZE:
```

Comment:
This might prevent the same address from having "a lot of nano transactions" in the same block.